aws/resource_aws_service_discovery_http_namespace_test.go

```diff
@@ -2,15 +2,96 @@ package aws
 
 import (
 	"fmt"
+	"log"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/servicediscovery/waiter"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_service_discovery_http_namespace", &resource.Sweeper{
+		Name: "aws_service_discovery_http_namespace",
+		F:    testSweepServiceDiscoveryHttpNamespaces,
+		Dependencies: []string{
+			"aws_service_discovery_service",
+		},
+	})
+}
+
+func testSweepServiceDiscoveryHttpNamespaces(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sdconn
+	var sweeperErrs *multierror.Error
+
+	input := &servicediscovery.ListNamespacesInput{
+		Filters: []*servicediscovery.NamespaceFilter{
+			{
+				Condition: aws.String(servicediscovery.FilterConditionEq),
+				Name:      aws.String(servicediscovery.NamespaceFilterNameType),
+				Values:    aws.StringSlice([]string{servicediscovery.NamespaceTypeHttp}),
+			},
+		},
+	}
+
+	err = conn.ListNamespacesPages(input, func(page *servicediscovery.ListNamespacesOutput, isLast bool) bool {
+		if page == nil {
+			return !isLast
+		}
+
+		for _, namespace := range page.Namespaces {
+			if namespace == nil {
+				continue
+			}
+
+			id := aws.StringValue(namespace.Id)
+			input := &servicediscovery.DeleteNamespaceInput{
+				Id: namespace.Id,
+			}
+
+			log.Printf("[INFO] Deleting Service Discovery HTTP Namespace: %s", id)
+			output, err := conn.DeleteNamespace(input)
+
+			if err != nil {
+				sweeperErr := fmt.Errorf("error deleting Service Discovery HTTP Namespace (%s): %w", id, err)
+				log.Printf("[ERROR] %s", sweeperErr)
+				sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+				continue
+			}
+
+			if output != nil && output.OperationId != nil {
+				if _, err := waiter.OperationSuccess(conn, aws.StringValue(output.OperationId)); err != nil {
+					sweeperErr := fmt.Errorf("error waiting for Service Discovery HTTP Namespace (%s) deletion: %w", id, err)
+					log.Printf("[ERROR] %s", sweeperErr)
+					sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+					continue
+				}
+			}
```

Comment:
The deletion logic is repeated here and in the resource itself. Would it make sense to have it in one place and call it for the resource and the sweeper?

Reply:
Most likely, yes. Ideally we can setup the sweepers to call the existing resource deletion logic rather than re-inventing the wheel each sweeper. Probably worth a followup design proposal though. 👍